core/src/main/java/io/questdb/griffin/UpdateOperatorImpl.java

```diff
@@ -145,6 +142,12 @@ public long executeUpdate(SqlExecutionContext sqlExecutionContext, UpdateOperati
                     final long currentRow = Rows.toLocalRowID(rowId);
 
                     if (rowPartitionIndex != partitionIndex) {
+                        if (tableWriter.isPartitionReadOnly(rowPartitionIndex)) {
+                            throw CairoException.critical(0)
```

Comment:
It seems weird that UPDATE over a RO partition returns an error while an INSERT does not. Both should return an error or silently ignore the given data. WDYT?

Reply:
the correct approach would be to raise an exception in both cases, however doing this would close any Line clients, and they would now need to modify clients to account for this situation. Instead, we are more permissive with inserts.
Updates cannot be done via Line, and the feedback is immediate, you cannot update a read-only partition.

Reply:
> however doing this would close any Line clients

We could simply introduce a new exception class and handle it in ILP code. This way errors would be correctly propagated in case of PGWire and only logged in case of ILP. WDYT?

Reply:
this is what I had, and we settled for this solution. 

I don't have strong feelings either way, but possibly we could address the need if it comes in a later PR?, this one starts to be moldy.,

Reply:
lets put it this way, updates are not a thing in time-series databases, we support them because some users require punctual changes here and there. So they will fail to update read-only partitions, loudly.

For the case of inserts, we chose this approach of ignoring inserts for read-only partitions (logging the fact at critical level).

Reply:
Current behavior feels inconsistent and counter-intuitive: most of our PGWire users would expect error propagation to kick in in case of a failed write. If that was discussed and agreed before on this PR, let's keep things as they're, but I'd expect to see some bug reports from users because of the behavior sooner or later.

Reply:
at present only partitions attached via symlink, explicitly done by the user, become read only and would be affected by this behaviour. And the documentation will say so, and the logs will show a C report.